hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/transaction/SimpleConcurrentFileWritesConflictResolutionStrategy.java

```diff
@@ -53,24 +56,45 @@ public class SimpleConcurrentFileWritesConflictResolutionStrategy
   @Override
   public Stream<HoodieInstant> getCandidateInstants(HoodieTableMetaClient metaClient, HoodieInstant currentInstant,
                                                     Option<HoodieInstant> lastSuccessfulInstant) {
+    boolean usesCompletionTimeOrdering = metaClient.getTableConfig().getTableVersion().greaterThanOrEquals(HoodieTableVersion.EIGHT);
     HoodieActiveTimeline activeTimeline = metaClient.getActiveTimeline();
     // To find which instants are conflicting, we apply the following logic
     // 1. Get completed instants timeline only for commits that have happened since the last successful write.
-    // 2. Get any scheduled or completed compaction that have started and/or finished after the current instant.
+    // 2. For tables with a version lower than 8, get any scheduled or completed compaction that have started and/or finished after the current instant.
+    //    - Note: this is not required for v8 tables which use completion time ordering of log files to build file groups.
     // 3. Get any completed replace commit that happened since the last successful write and any pending replace commit.
     // We need to check for write conflicts since they may have mutated the same files that are being newly created by the current write.
+    Predicate<HoodieInstant> compactionInstantFilter = (HoodieInstant instant) -> {
```

Comment:
Can we add a new strategy class and not mess up the v8 and v6.

Reply:
Do you mean a new implementation of `ConflictResolutionStrategy`? It would allow the user to potentially specify the class that does not map to their table version which can cause issues.

Reply:
One way to make the code cleaner is to have private methods for handling the different versions so this method just checks table version and then delegates to these methods. Then the javadocs per method can also contain the specifics for that table version.

Reply:
I've split this out into two internal methods which seems like the best way to ensure users automatically pick up these improvements if they upgraded to table version 8. The diff is also smaller now so it is more clear what is changing as part of this PR.